pramen/extras/src/main/scala/za/co/absa/pramen/extras/sink/EnceladusSink.scala

```diff
@@ -145,63 +151,140 @@ class EnceladusSink(sinkConfig: Config,
+  }
+
+  private[extras] def getBasePath(tableName: String,
+                                  options: Map[String, String]): Path = {
+    if (!options.contains(OUTPUT_PATH_KEY)) {
+      throw new IllegalArgumentException(s"$OUTPUT_PATH_KEY is not specified for Enceladus sink, table: $tableName")
+    }
+
+    new Path(options(OUTPUT_PATH_KEY))
+  }
+
+  private[extras] def getOutputPartitionPath(basePath: Path,
+                                             infoDate: LocalDate,
+                                             infoVersion: Int): Path = {
+    val partition = PartitionUtils.unpackCustomPartitionPattern(enceladusConfig.partitionPattern, enceladusConfig.infoDateColumn, infoDate, infoVersion)
+    new Path(basePath, partition)
+  }
+
+  private[extras] def writeToRawFolder(df: DataFrame,
+                                       recordCount: Long,
+                                       outputPartitionPath: Path)(implicit spark: SparkSession): Unit = {
+    val outputPathStr = outputPartitionPath.toUri.toString
+    log.info(s"Saving $recordCount records to the Enceladus raw folder: $outputPathStr")
+
+    val fsUtils = new FsUtils(spark.sparkContext.hadoopConfiguration, outputPathStr)
+    fsUtils.createDirectoryRecursiveButLast(outputPartitionPath)
+
+    val dfToWrite = enceladusConfig.recordsPerPartition match {
+      case Some(rpp) =>
+        val n = Math.max(1, Math.ceil(recordCount.toDouble / rpp)).toInt
+        log.info(s"Repartitioning to $n partitions...")
+        df.repartition(n)
+      case None      =>
+        df
+    }
+
+    dfToWrite.write
+      .mode(enceladusConfig.mode)
+      .format(enceladusConfig.format)
+      .options(enceladusConfig.formatOptions)
+      .save(outputPathStr)
+  }
+
+  private[extras] def generateInfoFile(df: DataFrame,
+                                       recordCount: Long,
+                                       outputPartitionPath: Path,
+                                       infoDate: LocalDate,
+                                       jobStart: Instant
+                                      )(implicit spark: SparkSession): Unit = {
+    if (enceladusConfig.generateInfoFile) {
+      InfoFileGeneration.generateInfoFile(enceladusConfig.pramenVersion,
+        enceladusConfig.timezoneId,
+        recordCount,
+        df,
+        outputPartitionPath,
+        infoDate,
+        jobStart,
+        jobStart)(spark, sinkConfig)
+    }
+  }
+
+  private[extras] def runEnceladus(tableName: String,
+                                   infoDate: LocalDate,
+                                   infoVersion: Int,
+                                   basePath: Path,
+                                   options: Map[String, String]): Unit = {
+    if (!options.contains(DATASET_NAME_KEY) || !options.contains(DATASET_VERSION_KEY)) {
+      log.info(s"Enceladus dataset name and/or version are not specified, skipping the Enceladus execution for $tableName.")
+      return
```

Comment:
Is this needed here? Seems like an extra

Reply:
Do you mean the logging or the condition?

Reply:
I mean the `return`. Also IIRC we tried avoid usage of the `return` keyword

Reply:
I avoid it too, but here the check is needed since DATASET_NAME_KEY and DATASET_VERSION_KEY are optional and might not be defined.

And I'd rather return earlier here than have a long else block.

Reply:
Last question. This is a small thing, but wouldn't it be easier (or more standard) to call the function conditionally than adding this?

Reply:
I was thinking about it. But I wanted the function to be responsible of knowing which are option keys that are needed for the execution. If the set of keys needed for Enceladus execution changes the single point of change will be this method. If I move the condition to the caller, then the caller needs to be changed too.
So this implementation is close to SRP.

If you are really annoyed by this return, I can split this method in 2. One checks the conditions and runs the other.

Reply:
Nono, I am fine with it. Thank you for taking the time to answer.